eng/common/pipelines/templates/steps/sparse-checkout.yml

```diff
@@ -22,84 +22,20 @@ steps:
-
-            if (Test-Path .git/info/sparse-checkout) {
-              $hasInitialized = $true
-              Write-Host "Repository $($repository.Name) has already been initialized. Skipping this step."
-            } else {
-              Write-Host "Repository $($repository.Name) is being initialized."
-
-              if ($repository.Commitish -match '^refs/pull/\d+/merge$') {
-                Write-Host "git clone --no-checkout --filter=tree:0 -c remote.origin.fetch='+$($repository.Commitish):refs/remotes/origin/$($repository.Commitish)' https://github.com/$($repository.Name) ."
-                git clone --no-checkout --filter=tree:0 -c remote.origin.fetch=''+$($repository.Commitish):refs/remotes/origin/$($repository.Commitish)'' https://github.com/$($repository.Name) .
-              } else {
-                Write-Host "git clone --no-checkout --filter=tree:0 https://github.com/$($repository.Name) ."
-                git clone --no-checkout --filter=tree:0 https://github.com/$($repository.Name) .
-              }
-
-              # Turn off git GC for sparse checkout. Note: The devops checkout task does this by default
-              Write-Host "git config gc.auto 0"
-              git config gc.auto 0
-
-              Write-Host "git sparse-checkout init"
-              git sparse-checkout init
-
-              # Set non-cone mode otherwise path filters will not work in git >= 2.37.0
-              # See https://github.blog/2022-06-27-highlights-from-git-2-37/#tidbits
-              Write-Host "git sparse-checkout set --no-cone '/*' '!/*/' '/eng'"
-              git sparse-checkout set --no-cone '/*' '!/*/' '/eng'
-            }
-
-            # Prevent wildcard expansion in Invoke-Expression (e.g. for checkout path '/*')
-            $quotedPaths = $paths | ForEach-Object { "'$_'" }
-            $gitsparsecmd = "git sparse-checkout add $quotedPaths"
-            Write-Host $gitsparsecmd
-            Invoke-Expression -Command $gitsparsecmd
-
-            Write-Host "Set sparse checkout paths to:"
-            Get-Content .git/info/sparse-checkout
-
-            # sparse-checkout commands after initial checkout will auto-checkout again
-            if (!$hasInitialized) {
-              # Remove refs/heads/ prefix from branch names
-              $commitish = $repository.Commitish -replace '^refs/heads/', ''
-
-              # use -- to prevent git from interpreting the commitish as a path
-              Write-Host "git -c advice.detachedHead=false checkout $commitish --"
-
-              # This will use the default branch if repo.Commitish is empty
-              git -c advice.detachedHead=false checkout $commitish --
-            } else {
-              Write-Host "Skipping checkout as repo has already been initialized"
-            }
-
-            Pop-Location
-        }
-
+        $req = Invoke-WebRequest "https://raw.githubusercontent.com/$(Build.Repository.Name)/main/eng/common/scripts/sparse-checkout.ps1"
```

Comment:
Should we add error handling for when this call fails (which it inevitably will)?  Or just let the exception bubble up?